### ISSUE TITLE
set max `pytorch-lightning` version to 2.1

### ISSUE THREAD

**Issue by ArneBinder** [ArneBinder/pytorch-ie#511]
from 2.2 onwards, the models are not automatically set to train mode during training (see section "Improved Handling of Evaluation Mode" in the [release notes](https://github.com/Lightning-AI/pytorch-lightning/releases/tag/2.2.0)). this affects e.g. loaded Huggingface transformer models

EDIT: maybe better to directly fix this by setting the model to train in the `train` method of the template? but that would require access to/knowledge of internals of `PyTorchIEModel` implementations (the HF model does not need to be at `.model`) from the template (would break the generality of the `train` method). Does `pytorch-lightning` provide functionally (recap that `PyTorchIEModel` is derived from lightning modules) that we could access from the template? Or do we need to add sth like a `set_train()` to the interface of `PyTorchIEModel` ourselfes?

fyi @RainbowRivey

**Comment by ArneBinder** [comment]
after discussion with @RainbowRivey: it should be sufficient to just call `model.train()` in `train.py` of the template before passing the model to the trainer

**Comment by ArneBinder** [comment]
This is implemented not here (in pytorch-ie), but in https://github.com/ArneBinder/pytorch-ie-hydra-template-1/pull/215.
